src/e2e-test/java/io/cdap/plugin/utils/PubSubClient.java

```diff
@@ -56,4 +92,76 @@ public static String getTopicCmekKey(String topicId) throws IOException {
     return getTopic(topicId).getKmsKeyName();
   }
 
+
+  public static void publishWithErrorHandlerExample(String projectId, String topicId, List<String> dataMessages)
+      throws IOException, InterruptedException {
+    TopicName topicName = TopicName.of(PluginPropertyUtils.pluginProp(ConstantsUtil.PROJECT_ID), topicId);
+    Publisher publisher = null;
+    try {
+      publisher = Publisher.newBuilder(topicName).build();
+
+//      List<String> messages = Arrays.asList("first message", "second message");
+      for (final String message : dataMessages) {
+        ByteString data = ByteString.copyFromUtf8(message);
+        PubsubMessage pubsubMessage = PubsubMessage.newBuilder().setData(data).build();
+        ApiFuture<String> future = publisher.publish(pubsubMessage);
+
+        // Adding an asynchronous callback to handle success / failure
+        ApiFutures.addCallback(
+            future,
+            new ApiFutureCallback<String>() {
+
+              @Override
+              public void onFailure(Throwable throwable) {
+                if (throwable instanceof ApiException) {
+                  ApiException apiException = ((ApiException) throwable);
+                   // details on the API exception
+                   System.out.println(apiException.getStatusCode().getCode());
```

Comment:
Remove and use logger. Cross check and remove wherever present.

Reply:
removed from everywhere